proof/refine/RISCV64/Invariants_H.thy

```diff
@@ -51,66 +71,82 @@ definition pspace_no_overlap' :: "obj_ref \<Rightarrow> nat \<Rightarrow> kernel
      \<lambda>s. \<forall>x ko. ksPSpace s x = Some ko \<longrightarrow>
                 (mask_range x (objBitsKO ko)) \<inter> {ptr .. (ptr && ~~ mask bits) + mask bits} = {}"
 
-definition ko_wp_at' :: "(kernel_object \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
+definition
   "ko_wp_at' P p s \<equiv> \<exists>ko. ksPSpace s p = Some ko \<and> is_aligned p (objBitsKO ko) \<and> P ko \<and>
-                           ps_clear p (objBitsKO ko) s"
+                           ps_clear p (objBitsKO ko) s \<and> objBitsKO ko < word_bits"
+
+lemma valid_sz_simps:
+  "objBitsKO ko < word_bits =
+    (case ko of
+      (KOSchedContext sc) \<Rightarrow> scBitsFromRefillLength sc < word_bits
+      | _ \<Rightarrow>    True)"
+  by (cases ko;
+      clarsimp simp: objBits_def objBitsKO_def word_size_def archObjSize_def pageBits_def word_bits_def
+                     tcbBlockSizeBits_def epSizeBits_def ntfnSizeBits_def cteSizeBits_def word_size
+                     pteBits_def wordSizeCase_def wordBits_def replySizeBits_def
+              split: arch_kernel_object.splits)
 
-definition obj_at' :: "('a::pspace_storable \<Rightarrow> bool) \<Rightarrow> machine_word \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  obj_at'_real_def:
-  "obj_at' P p s \<equiv> ko_wp_at' (\<lambda>ko. \<exists>obj. projectKO_opt ko = Some obj \<and> P obj) p s"
+definition
+  obj_at' :: "('a::pspace_storable \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool"
+where obj_at'_real_def:
```

Comment:
Probably also part of the automated patch bit, but please restore the original correct style (name on same line as `definition` and `where` on same line as type when it fits (and all on same line if possible))

Reply:
In fact, is there any actual change in this one? It's hard to spot.